src/plugins/intel_gpu/src/plugin/ops/pad.cpp

```diff
@@ -18,21 +18,53 @@ static void CreatePadOp(Program& p, const std::shared_ptr<ngraph::op::v1::Pad>&
     auto inputs = p.GetInputInfo(op);
     std::string layerName = layer_type_name_ID(op);
 
+    std::vector<cldnn::input_info> new_inputs = {inputs[0]};
+
+    auto pads_begin_constant = std::dynamic_pointer_cast<ngraph::op::v0::Constant>(op->input_value(1).get_node_shared_ptr());
+    std::vector<int64_t> pads_begin = std::vector<int64_t>{};
+    if (pads_begin_constant) {
+        pads_begin = pads_begin_constant->cast_vector<int64_t>();
+    } else {
+        new_inputs.push_back(inputs[1]);
+    }
+
+    auto pads_end_constant = std::dynamic_pointer_cast<ngraph::op::v0::Constant>(op->input_value(2).get_node_shared_ptr());
+    std::vector<int64_t> pads_end = std::vector<int64_t>{};
+    if (pads_end_constant) {
+        pads_end = pads_end_constant->cast_vector<int64_t>();
+    } else {
+        new_inputs.push_back(inputs[2]);
+    }
+
     float pad_value = 0.f;
+    bool pad_value_input_constant = false;
     if (op->get_input_size() == 4) {
         auto const_node = std::dynamic_pointer_cast<ngraph::op::v0::Constant>(op->get_input_node_shared_ptr(3));
-        OPENVINO_ASSERT(const_node, "Unsupported const node type in ", op->get_friendly_name(), " (", op->get_type_name(), ")");
-        ov::op::util::get_single_value(const_node, pad_value);
+        if (const_node) {
+            ov::op::util::get_single_value(const_node, pad_value);
+            pad_value_input_constant = true;
+        }
     }
 
-    auto tilePrim = cldnn::border(layerName,
-                                  inputs[0],
-                                  op->get_pads_begin(),
-                                  op->get_pads_end(),
-                                  op->get_pad_mode(),
-                                  pad_value);
+    if (!pad_value_input_constant && op->get_pad_mode() == ov::op::PadMode::CONSTANT) {
+        new_inputs.push_back(inputs[3]);
+        auto tilePrim = cldnn::border(layerName,
+                                    new_inputs,
+                                    pads_begin,
+                                    pads_end,
+                                    cldnn::padding());
+
+        p.add_primitive(*op, tilePrim);
+    } else {
+        auto tilePrim = cldnn::border(layerName,
+                                    new_inputs,
+                                    pads_begin,
+                                    pads_end,
+                                    op->get_pad_mode(),
+                                    pad_value);
 
-    p.add_primitive(*op, tilePrim);
+        p.add_primitive(*op, tilePrim);
+    }
```

Comment:
Can't we just set some flag with bit position which presents whether the input is const or not? 
E.g.,
1. declare size_t const_inputs = 0x0;  // wich will represent {pad_begin, pad_end, pad_value}
2. if pad_begin is const 
     const_inputs |= (0x1 << 2);
    if pad_end is const 
     const_inputs |= (0x1 << 1);
    if pad_value is const 
     const_inputs |= 0x1;

And in cldnn program node, we can parse it like 
bool pad_begin_is_const = const_inputs & (0x1 << 2);
bool pad_end_is_const = const_inputs & (0x1 << 1);
bool pad_val_is_const = const_inputs & (0x1);

If we just set this flag in creation of the primitive, then we dont need this complicated branches.

Reply:
Of course we can improve readability using enum for each position, like
enum PAD_CONST_INPUT {
    PAD_BEGIN_CONST = ( 0x1 << 2),
...
}
const inputs |= PAD_BEGIN_CONST;

Reply:
Done. Used PAD_NON_CONST_INPUT to make default 0 for all constant inputs.